Improved and Added Documentation

atulim opened the pull request:
While looking at the code, I observed obsolete documentation patterns which did not confine to the Documentation Guidelines.
I changed the pattern of commenting and also added a comment for the logger.
----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

matthiasgeiger commented:
Hi @atulim,
first, thank you for your interest in contributing to the JabRef project!

However, please consider to provide only PRs with a real benefit for our users. The project team as only limited resources to review and give feedback to contribution, so we decided not to accept pure refactoring and code improvement pull requests.

Please have a look at the issues labeled as [`good-first-issue`](https://github.com/JabRef/jabref/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) to find suitable issues to start working on.

And also have a look at [our contribution guidelines](https://github.com/JabRef/jabref/blob/master/CONTRIBUTING.md).

Thank you!

atulim commented:
> Hi @atulim,
> first, thank you for your interest in contributing to the JabRef project!
> 
> However, please consider to provide only PRs with a real benefit for our users. The project team as only limited resources to review and give feedback to contribution, so we decided not to accept pure refactoring and code improvement pull requests.
> 
> Please have a look at the issues labeled as [`good-first-issue`](https://github.com/JabRef/jabref/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22) to find suitable issues to start working on.
> 
> And also have a look at [our contribution guidelines](https://github.com/JabRef/jabref/blob/master/CONTRIBUTING.md).
> 
> Thank you!

@matthiasgeiger

When I was reading the code, I found out that the codes had little to no documentation. Every open source code should have proper documentation. For the sole purpose only, I added some documentation and CORRECTED some issues.
Yes. I have contributed to the issues labeled as good-first-issue. In response to providing real benefit to users, we cannot provide wrong and irrelevant message. For that purpose, look at my second commit where you had previously provided wrong error message.
Kindly reply to this message and look at commits please. I would add substantial changes in next few days for U.I.

Thank you.

matthiasgeiger commented on src/main/java/org/jabref/JabRefException.java:
```diff
@@ -3,11 +3,14 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+// Class for handling exceptions in GUI and printing corresponding exception messages
```
Not a Javadoc class comment. No real added value by this comment: as this class is extending `Exception` it's purpose is obvious.

atulim replied:
Okay.
I understand this.

matthiasgeiger commented on src/main/java/org/jabref/JabRefException.java:
```diff
@@ -3,11 +3,14 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(JabRefException.class);
     private String localizedMessage;
 
+    // Passing the exception message to the base class
```
Not a Javadoc comment. Obvious purpose of the constructor as this is the usual pattern for exception classes in Java.

atulim replied:
Okay.
I understand this. I will try to fix the said issues as soon as possible. 

Thank you.

matthiasgeiger commented on src/main/java/org/jabref/JabRefException.java:
```diff
@@ -29,11 +32,13 @@ public JabRefException(String message, String localizedMessage, Throwable cause)
     @Override
     public String getLocalizedMessage() {
+        // If there is no localized message
         if (localizedMessage == null) {
```
No added value as if clause is self-explanatory.

atulim replied:
Okay sir.

matthiasgeiger commented on src/main/java/org/jabref/JabRefMain.java:
```diff
@@ -31,11 +31,10 @@
-/**
- * JabRef MainClass
- */
+// JabRef MainClass
```
Why do you think this is better than the Javadoc comment?

atulim replied:
Some of the large organisations use '//' to mark all the comments whether it be Doc comment or something else.
The main purpose is the reduction of three lines into one lines which proves beneficial for their large amount of code.
If your organisation prefers this pattern, I have no issue.

matthiasgeiger commented on src/main/java/org/jabref/JabRefException.java:
```diff
@@ -29,11 +32,13 @@ public JabRefException(String message, String localizedMessage, Throwable cause)
     }
-
+     
+    // To display localized message on error
```
No comment needed as already defined in `Exception` base class.

atulim replied:
okay.

matthiasgeiger commented on src/main/java/org/jabref/JabRefException.java:
```diff
@@ -3,11 +3,14 @@
 public class JabRefException extends Exception {
-
+    
+    // Creating a logger and using it to write messages to log files
```
Not necessary as naming of variable is already self-explanatory. "write to logfiles" depends on log4j configuration that is maintained elsewhere - thus: obsolete and misleading comment here.

atulim replied:
I personally faced a problem of googling log4j. For the sole purpose I wrote this because it is not common for students to use logger in their JavaFx project thus they do not know about this.

matthiasgeiger commented on src/main/java/org/jabref/JabRefMain.java:
```diff
@@ -31,11 +31,10 @@
+// JabRef MainClass
 public class JabRefMain extends Application {
 
+    // Creating a logger and using it to write messages to log files
```
See above.

atulim replied:
Thank you.

LinusDietz commented:
In the devcall we generally agreed to @matthiasgeiger. We honor the effort to improve stuff, but we only see the fixing of the typos as an improvement. So please revert the changes regarding new comments, etc. within a week (that is until next Friday, April, 5) so we can merge this PR.

atulim commented:
Reverted back the changes. Ready for merge.
Thank you.
